### PR TITLE
Client side for charts

### PR DESCRIPTION
**Pull request by Open-Fuse**
Other commits from my other PR

Switched requests for data to post requests
Gets the user's id server-side and send it in requests

Needs to have changes from the master to properly test

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/ChartsDataServlet.java [hunk 1]
```diff
@@ -1,132 +1,187 @@
...
+		private double footprint;
+		
+		private FormattedMeal(Date date, double footprint) {
 			this.date = date;
+			this.footprint = footprint;
 		}
 	}
 	
 	/*
 	 * Generates Dummy data to be sent in GET requests
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/servlets/ChartsDataServlet.java]: This comment looks outdated now right? This `Init` method isn't generating any dummy data now.

File: src/main/java/com/google/codeu/servlets/ChartsDataServlet.java [hunk 2]
```diff
@@ -1,132 +1,187 @@
...
+		Instant now = Instant.now();
+		Instant before = now.minus(Duration.ofDays(7));
+		Date sevenDaysAgo = Date.from(before);
 		
-		return gson.toJson(averageCarbonPerDay).toString();
+		FilterPredicate dateFilter = new Query.FilterPredicate("date", Query.FilterOperator.GREATER_THAN, sevenDaysAgo);
+		FilterPredicate userFilter = new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, user);
+		
+		Query query = new Query("UserMeal")
+							.setFilter(new Query.CompositeFilter(Query.CompositeFilterOperator.AND, Arrays.asList(dateFilter, userFilter)))
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/servlets/ChartsDataServlet.java]: The spacing here looks a bit wonky

**Review by NickChapman — COMMENTED** [review]
Couple small things, otherwise lookin' really good! Can't wait to see it in action!

File: src/main/java/com/google/codeu/servlets/ChartsDataServlet.java (hunk 2 above)
Comment (Ivyzhang1998) [reply src/main/java/com/google/codeu/servlets/ChartsDataServlet.java]: UserMeal -> EatenMeal

File: src/main/java/com/google/codeu/servlets/FormHandlerServlet.java [hunk 3]
```diff
@@ -172,6 +172,11 @@ private void handleEatenFood(HttpServletRequest request, HttpServletResponse res
         if (blobKey == null) {
             out.println("<p>You ate " + meal.getAmount() + " of "  + meal.getFood() +
                     " on " + format.format(meal.getDate()) + ".</p>");
+            if (meal.getCO2() == 0){
+               out.println("This Food is currently not in our Database. Feel free to add its CO2 information to our CO2 Data Form!");}
```
Comment (dayhahm) [inline src/main/java/com/google/codeu/servlets/FormHandlerServlet.java:176]: would it be possible to add a link to the form? alternatively I can just add it in for my pr later

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 4]
```diff
@@ -16,16 +16,20 @@
...
             <div class="hero hero-sm bg-dark">
                 <div class="hero-body">
                     <h1>FoodO2</h1>
                     <p>By The Carbon Mob</p>
                 </div>
             </div>
 
+            <p>Welcome! This is our FoodCO2 APP, where you can calculate the carbon footprint of your diet
+      and keep track of how much greenhouse gas emissions you contribute through your consumption.
+      Overall, we hope that this APP will raise awarenss about the environmental impact of different food choices and encourage
```
Comment (dayhahm) [inline src/main/webapp/WEB-INF/views/index.jsp:30]: awarenss --> awareness

**Review by dayhahm — APPROVED** [review]
couple of small changes, but they're all optional so approving! this looks amazing!!

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 4 above)
Comment (Open-Fuse) [reply src/main/webapp/WEB-INF/views/index.jsp:30]: Weird this isn't even from my commits, It's from a rebase of Ivy's changes

**Pull request merged**